net/packet/ip6.go

```diff
@@ -5,30 +5,109 @@
 package packet
 
 import (
+	"encoding/binary"
 	"fmt"
 
 	"inet.af/netaddr"
 )
 
 // IP6 is an IPv6 address.
-type IP6 [16]byte
+type IP6 struct {
+	Hi, Lo uint64
```

Comment:
was this faster than [16]byte?

Reply:
Significantly so yes, because it makes the implementation of net6.Contains way faster, and that's all we do with this type. Also, we generally end up comparing the full 128 bits, whereas the netaddr implementation is balanced for being faster with short prefixes (we do bytewise comparison and stop at the mask size).

Might be interesting to see if we can make netaddr this implementation as well and still retain speed for all the more generic ops that it supports.

See benchstats in https://github.com/tailscale/tailscale/pull/918/commits/0432ab12c7451967bdfa219d2353c1ca758af86c